File: GXpert/GXpert.Web/Modules/Skills/Skill/SkillForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Skills.Forms;

[FormScript("Skills.Skill")]
[BasedOnRow(typeof(SkillRow), CheckNames = true)]
public class SkillForm
{
    public int SkillCategoryId { get; set; }
    public string Name { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
    
   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Skills/SkillForm.ts
import { StringEditor, TextAreaEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SkillForm {
    Name: StringEditor;
    Description: TextAreaEditor;
    SkillCategoryId: LookupEditor;
}

export class SkillForm extends PrefixedContext {
    static readonly formKey = 'Skills.Skill';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SkillForm.init)  {
            SkillForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = LookupEditor;

            initFormType(SkillForm, [
                'Name', w0,
                'Description', w1,
                'SkillCategoryId', w2
            ]);
        }
    }
}